engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -268,10 +269,25 @@ class TableScan : public kernel {
 	virtual kstatus run() {
 		CodeTimer timer;
 
-		while( input.wait_for_next() ) {
-			auto batch = input.next();
-			this->add_to_output_cache(std::move(batch));
+		int TABLE_SCAN_KERNEL_NUM_THREADS = 1;
+		std::map<std::string, std::string> config_options = context->getConfigOptions();
+		auto it = config_options.find("TABLE_SCAN_KERNEL_NUM_THREADS");
+		if (it != config_options.end()){
+			TABLE_SCAN_KERNEL_NUM_THREADS = std::stoi(config_options["TABLE_SCAN_KERNEL_NUM_THREADS"]);
 		}
+
+		std::vector<BlazingThread> threads;
+		for (int i = 0; i < TABLE_SCAN_KERNEL_NUM_THREADS; i++) {
```

Comment:
This is alot more overhead than we need. If th elimit is 4 for example we shouldn't have 4 threads running at the same time but rather have up to 4 .next() being called at the same time and limiting another .next() from calling until one of  of them finishes. This is alot more overhead than the pattern having a single thread that launches load_batch as a seperate thread that can finish whenever it needs to

Reply:
i dont understand what overhead you are talking about. It sounds like you are talking about creating a different thread for every file. THat does not sound efficient